packages/api/amplify_api/lib/src/decorators/authorize_http_request.dart

```diff
@@ -24,11 +25,13 @@ Future<AWSBaseHttpRequest> authorizeHttpRequest(
   AWSBaseHttpRequest request, {
   required AWSApiConfig endpointConfig,
   required AmplifyAuthProviderRepository authProviderRepo,
+  APIAuthorizationType? authorizationMode,
 }) async {
-  if (request.headers.containsKey(AWSHeaders.authorization)) {
+  if (request.headers.containsKey(AWSHeaders.authorization) ||
+      request.headers.containsKey(xApiKey)) {
```

Comment:
Is this correct? What if the user's lambda wants to repurpose this header? Probably need to check that `authType == api key` as well.

Reply:
I will add check for `== apiKey`.